pkg/process/procutil/process_linux.go

```diff
@@ -212,10 +223,17 @@ func (p *Probe) ProcessesByPID(now time.Time) (map[int32]*Process, error) {
 			continue
 		}
 
-		statusInfo := p.parseStatus(pathForPID)
-		statInfo := p.parseStat(pathForPID, pid, now)
-		memInfoEx := p.parseStatm(pathForPID)
+		var (
+			statusInfo = &statusInfo{}
+			statInfo   = &statInfo{}
+			memInfoEx  = &MemoryInfoExStat{}
+		)
 
+		if p.collectStats {
```

Comment:
This only takes care of it for Linux.

Reply:
Yep I still have to set it up for windows. Darwin however, may not be possible in a single due to it's dependency on gopsutil. I will continue to investigate this

Reply:
Alright, I just added some basic support for windows metadata collection!